tests/framework/e2e/etcdctl.go

```diff
@@ -507,3 +508,60 @@ func (ctl *EtcdctlV3) UserChangePass(user, newPass string) error {
 	_, err = cmd.Expect("Password updated")
 	return err
 }
+
+func (ctl *EtcdctlV3) Watch(ctx context.Context, key string, opts config.WatchOptions) clientv3.WatchChan {
+	args := ctl.cmdArgs()
+	args = append(args, "watch", key)
+	if opts.RangeEnd != "" {
+		args = append(args, opts.RangeEnd)
+	}
+	args = append(args, "-w", "json")
+	if opts.Prefix {
+		args = append(args, "--prefix")
+	}
+	if opts.Interactive {
+		args = append(args, "--interactive")
+	}
+	if opts.Revision != 0 {
+		args = append(args, "--rev", fmt.Sprint(opts.Revision))
+	}
+	if len(opts.ExecCmd) > 0 {
+		args = append(args, "--")
+		args = append(args, opts.ExecCmd...)
+	}
+
+	proc, err := SpawnCmd(args, nil)
+	if err != nil {
+		return nil
+	}
+	defer proc.Stop()
```

Comment:
This will kill the proc even though we started separate read goroutine on line 545